Lib/test/test_json/test_tool.py

```diff
@@ -206,3 +207,14 @@ def test_ensure_ascii_default(self):
         # asserting an ascii encoded output file
         expected = [b'{', rb'    "key": "\ud83d\udca9"', b"}"]
         self.assertEqual(lines, expected)
+
+    @unittest.skipIf(sys.platform =="win32", "The test is failed with ValueError on Windows")
+    def test_broken_pipe_error(self):
+        cmd = [sys.executable, '-m', 'json.tool']
+        proc = subprocess.Popen(cmd,
+                                stdout=subprocess.PIPE,
+                                stdin=subprocess.PIPE)
```

Comment:
Please add a comment explaining that stdin=PIPE ensures that the stdout close is closed before json.tool attempts to write into stdout.

It's a nice synchronization primite: better than time.sleep() ;-)